intranet/utils/locking.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, Iterable, List, Union
+
+from django.db.models import Manager, Model, QuerySet
+
+
+def lock_on(items: Iterable[Union[Model, Manager, QuerySet]]) -> None:
+    """Given an iterable of ``Model`` instances, ``Manager``s, and/or ``QuerySet``s, locks the
+    corresponding database rows.
+
+    More specifically, this uses the Django ORM's ``select_for_update()`` method, which translates
+    to a ``SELECT FOR UPDATE`` SQL query. For more information on what this actually does in
+    PostgreSQL (used as the database backend in dev environments and in production) see PostgreSQL's
+    documentation on locking at
+    https://www.postgresql.org/docs/8.3/explicit-locking.html#LOCKING-ROWS.
```

Comment:
```suggestion
    https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-ROWS.
```
We should not necessarily pin documentation to a specific PostgreSQL version.